### ISSUE TITLE
Floating scroll not updating when container is hidden

### ISSUE THREAD

**Issue by hngu** [Amphiluke/floating-scroll#7]
Hi there!

I have been using your plugin and find it to be very useful! But I noticed an issue where:
- I have two tabs, where each tab has a table that contains a floating scroll instance
- When user loads, they view the first tab with a table, but there is no visible floating scroll because the table is not very wide.
- If the user goes to the second tab and sees a very wide table, the floating scroll shows up.
- If the user goes back to the first tab, the floating scroll from the second tab is still visible.
- Only when I resize the floating scroll disappears.

The problem is that I cannot detect when an element visibility changes and then ask the floating scroll to update. The way I fixed this was to modify your plugin from this:

```
        var flscroll = $("<div class='fl-scrolls'></div>");
        $("<div></div>").appendTo(flscroll).css({width: this.cont.block.scrollWidth + "px"});
        return flscroll.appendTo('body');
```

to this: 

```
        var flscroll = $("<div class='fl-scrolls'></div>");
        $("<div></div>").appendTo(flscroll).css({width: this.cont.block.scrollWidth + "px"});
        return flscroll.appendTo($(this.cont.block));
```

Do you think this is a better approach since now the floating scroll instances are attached to the container and not the body?

Thanks!

**Comment by Amphiluke** [comment]
Hi!
Thank you for this detailed description of the problem. The solution you proposed looks clean and smart. I'm going to update the plugin within the next weekend. This doesn't appear to be a large rework but may potentially cause small CSS issues (owing to inheritance), so I want to test carefully.

Thanks again, and have a nice day!

**Comment by hngu** [comment]
Awesome! Yes let me know how the testing goes. This was just a random suggestion but I do hope it works out. Thanks!

**Comment by Amphiluke** [comment]
A fix for the issue is released in [v2.2.5](https://github.com/Amphiluke/floating-scroll/releases/tag/v2.2.5). Widget instances are now attached to the container they serve.

@hngu thanks both for the report and for the working solution.
